src/general/ReversibleAction.cs

```diff
@@ -12,6 +12,13 @@ public abstract class ReversibleAction
     [JsonProperty]
     public bool Performed { get; private set; }
 
+    /// <summary>
+    ///   A sub-action is an action that was automatically performed with another action.
+    ///   Sub-actions should not be noticeable by the user.
+    /// </summary>
+    [JsonIgnore]
+    public abstract bool IsSubAction { get; }
```

Comment:
I kinda disagree with this subaction design (related to my previous point). Instead I think something like action inheritance or composition should be used to make a single action encapsulate more things than it normally would.